src/MongoDB/ReadPreference.c

```diff
@@ -343,6 +369,21 @@ static PHP_METHOD(ReadPreference, getMode)
 	RETURN_LONG(mongoc_read_prefs_get_mode(intern->read_preference));
 } /* }}} */
 
+/* {{{ proto string MongoDB\Driver\ReadPreference::getModeString()
+   Returns the ReadPreference mode as string */
+static PHP_METHOD(ReadPreference, getModeString)
+{
+	php_phongo_readpreference_t* intern;
+
+	intern = Z_READPREFERENCE_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	PHONGO_RETURN_STRING(php_phongo_readpreference_get_mode_string(mongoc_read_prefs_get_mode(intern->read_preference)));
```

Comment:
I would be more concerned about passing `NULL` to `strlen()` in the `ZVAL_STRING` macro:

 * http://lxr.jpauli.tech/source/xref/php-5.6/Zend/zend_API.h#ZVAL_STRING
 * http://lxr.jpauli.tech/source/xref/php-7.4/Zend/zend_API.h#ZVAL_STRING

I believe `strlen(NULL)` is undefined in most implementations:

 * https://stackoverflow.com/q/5796103/162228
 * https://en.cppreference.com/w/c/string/byte/strlen

In practice, I don't think it's possible for the read preference to have a different mode unless it was added to the enum and the PHP driver was not updated to handle it. That said, a defensive solution may be to simply throw an exception here if `php_phongo_readpreference_get_mode_string` returns `NULL` and include the numeric code in the message.

Reply:
I was just reminded of https://github.com/mongodb/mongo-php-driver/pull/1061/files#r345720246. For consistency, perhaps you'd rather throw from `php_phongo_readpreference_get_mode_string` instead of `getModeString()`?